clients/python/tests/test_client.py

```diff
@@ -109,6 +109,67 @@ async def test_get(client: ModelRegistry):
     assert ma.id == _ma.id
 
 
+def test_get_registered_models(client: ModelRegistry):
+    models = 21
+
+    for name in [f"test_model{i}" for i in range(models)]:
+        client.register_model(
+            name,
+            "s3",
+            model_format_name="test_format",
+            model_format_version="test_version",
+            version="1.0.0",
+        )
+
+    rm_iter = client.get_registered_models().limit(10)
```

Comment:
```suggestion
    rm_iter = client.get_registered_models().limit(10)
    i = 0
    for model in rm_iter:
        assert model.name == f"test_model{i}"
        i += 1

    rm_iter.restart()
```

Just to highlight the standand, plain use of the iterator, I'd add similarly, with the value added it would cover for the reset case. Wdyt ?

I'd add similarly for the ModelVersion(s).

Then this looks good to me, thank you!
In subsequent PR expanding on test coverage for the different types of ordering.